Core vf conditions

isubasinghe opened the pull request.

isubasinghe committed:
add more call conditions to function calls

Commit:
fixed type signatures for call conditions

Commit:
fix: added wf_MsgInfo

Commit:
added rv conditions

Commit:
fix: rename system call postconditions

Commit:
fix: ensure reading from oracle

Commit:
completed post conds for replyrecv

Commit:
finished post conds for recv

math2001 committed:
Make assert node proof obligation jump to error node

Signed-off-by: Mathieu PATUREL <z5312157@unsw.edu.au>

math2001 committed:
Arbitrary variables for function calls need more work

unification is the thing that requires more care, see use_modified_ghost_using_prelude

Commit:
added empty fns for handle loop condition

math2001 committed:
Massive hack to make function calls work

math2001 committed:
Increase confidence in function call hack correctness

Signed-off-by: Mathieu PATUREL <z5312157@unsw.edu.au>

math2001 commented on ghost_data.py:
```diff
@@ -102,6 +137,40 @@ def ret_value(v: source.ExprVarT[source.VarNameKind]) -> source.ExprVarT[source.
+
+    # TODO: is this correct?
+    return conjs(
+        sle(label, label_maxval),
+        sle(count, count_maxval)
```
`sle` stands for `signed less equals`. I don't think signed comparison is what you mean to do here.

math2001 committed:
Define C_msg_info=SMT_msg_info in prelude

More mathematical style predicates

math2001 committed:
Rewrote post conditions for correspondence_(reply)?recv

Still need to encode wf_MsgInfo

It wasn't correct. We now use helper function to ensure we don't miss
any cases when copying composite types and doing case splitting.

Signed-off-by: Mathieu PATUREL <z5312157@unsw.edu.au>

math2001 committed:
Properly encode msginfo, which allows us to not even have to emit the wf_MsgInfo

If we had time, it'd still write it, but it's quite a bit of effort, for
not much reward.

It also makes the platform context smaller :)

Signed-off-by: Mathieu PATUREL <z5312157@unsw.edu.au>

math2001 committed:
commit global_smt_variables.py

isubasinghe committed:
feat: add mem-acc

isubasinghe committed:
ghost data fixes

isubasinghe committed:
reverted colour change

isubasinghe committed:
added handler loop ubc invariants
